File: app/src/main/java/com/prey/json/actions/PrivateIp.java
package com.prey.json.actions;

/**
 * Created by oso on 24-08-15.
 */

import java.util.HashMap;
import java.util.List;

import org.json.JSONObject;

import android.content.Context;

import com.prey.PreyLogger;
import com.prey.PreyPhone;
import com.prey.actions.HttpDataService;
import com.prey.actions.observer.ActionResult;
import com.prey.json.JsonAction;

public class PrivateIp  extends JsonAction{


    public List<HttpDataService> report(Context ctx, List<ActionResult> list, JSONObject parameters) {
        List<HttpDataService> listResult=super.report(ctx, list, parameters);
        return listResult;
    }

    public  List<HttpDataService> get(Context ctx, List<ActionResult> list, JSONObject parameters) {
        PreyLogger.d("Ejecuting PrivateIp Data.");
        List<HttpDataService> listResult=super.get(ctx, list, parameters);
        return listResult;
    }


    public HttpDataService run(Context ctx, List<ActionResult> lista, JSONObject parameters){
        PreyPhone phone=new PreyPhone(ctx);
        HttpDataService data = new HttpDataService("private_ip");
        HashMap<String, String> parametersMap = new HashMap<String, String>();

        String privateIp=phone.getWifi().getIpAddress();
        parametersMap.put(privateIp,privateIp);
        PreyLogger.d("privateIp:"+privateIp);;

        data.setSingleData(privateIp);

        return data;
    }


}

File: app/src/main/java/com/prey/json/actions/kotlin/PrivateIp.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.json.actions.kotlin

import android.content.Context
import com.prey.actions.kotlin.HttpDataService
import com.prey.actions.observer.kotlin.ActionResult
import com.prey.json.kotlin.JsonAction
import com.prey.kotlin.PreyLogger
import com.prey.kotlin.PreyPhone
import org.json.JSONObject

class PrivateIp : JsonAction() {
    override fun report(
        ctx: Context,
        list: MutableList<ActionResult>?,
        parameters: JSONObject?
    ): List<HttpDataService>? {
        val listResult = super.report(ctx, list, parameters)
        return listResult
    }

    override fun get(
        ctx: Context,
        list: MutableList<ActionResult>?,
        parameters: JSONObject?
    ): List<HttpDataService>? {
        PreyLogger.d("Ejecuting PrivateIp Data.")
        val listResult = super.get(ctx, list, parameters)
        return listResult
    }

    override fun run(
        ctx: Context,
        list: MutableList<ActionResult>?,
        parameters: JSONObject?
    ): HttpDataService? {
        val phone = PreyPhone(ctx)
        val data = HttpDataService("private_ip")
        val parametersMap = HashMap<String, String?>()
        val privateIp = phone.wifi!!.ipAddress!!
        parametersMap[privateIp] = privateIp
        PreyLogger.d("privateIp:$privateIp")
        data.setSingleData (privateIp)
        return data
    }
}